app/(dashboard)/dashboard/page.tsx

```diff
@@ -33,20 +40,43 @@ export default function DashboardPage() {
               },
             }))
           );
-          setError(null);
         }
+
+        if ("error" in gitlabData) {
+          setError((prevError) => prevError ? `${prevError}\n${gitlabData.error}` : gitlabData.error as string);
+          setMergeRequests([]);
+        } else {
+          console.log(gitlabData);
+          setMergeRequests(
+            gitlabData.map((mr) => ({
+              ...mr,
+              source: 'gitlab',  // Add this line
+              repository: {
+                id: mr.repoId,
+                name: mr.repo,
+                full_name: `${mr.owner}/${mr.repo}`,
+                owner: {
+                  login: mr.owner,
+                },
+              },
+            }))
+          );
+        }
+
+        setError(null);
         setLoading(false);
       } catch (error) {
-        console.error("Error fetching pull requests:", error);
+        console.error("Error fetching requests:", error);
         setLoading(false);
         setError(
-          "Failed to fetch pull requests. Please reconnect your GitHub account."
+          "Failed to fetch pull requests and merge requests. Please reconnect your GitHub and GitLab accounts."
```

Comment:
Makes the UX worse for someone who doesn't care about GitLab. Almost everyone uses GitHub so not super interested in this if it hurts the UX